libkbfs/tlf_journal.go

```diff
@@ -948,52 +930,37 @@ func (j *tlfJournal) doOnMDFlush(ctx context.Context,
 			rmds.MD.RevisionNumber())
 	}
 
-	// Remove saved blocks in chunks to avoid starving foreground file
-	// system operations that need the lock for too long.
-	var lastToRemove journalOrdinal
-	for {
-		nextLastToRemove, removedBytes, removedFiles, err := func() (journalOrdinal, int64, int64, error) {
-			j.journalLock.Lock()
-			defer j.journalLock.Unlock()
-			if err := j.checkEnabledLocked(); err != nil {
-				return 0, 0, 0, err
-			}
-
-			storedBytesBefore := j.blockJournal.getStoredBytes()
-
-			nextLastToRemove, removedBytes, removedFiles, err :=
-				j.blockJournal.onMDFlush(
-					ctx, maxSavedBlockRemovalsAtATime,
-					rmds.MD.RevisionNumber(), lastToRemove)
-			if err != nil {
-				return 0, 0, 0, err
-			}
+	blockJournal, err := func() (*blockJournal, error) {
+		j.journalLock.Lock()
+		defer j.journalLock.Unlock()
+		if err := j.checkEnabledLocked(); err != nil {
+			return nil, err
+		}
+		return j.blockJournal, nil
+	}()
+	if err != nil {
+		return err
+	}
 
-			storedBytesAfter := j.blockJournal.getStoredBytes()
+	// onMDFlush() only needs to be called under the flushLock, not
```

Comment:
Having to iterate through the gc journal while holding the lock (step 3) hurts some of the foreground performance benefits this was aiming for.  That said, I didn't actually see much improvement when I tested it.  However, this PR still cleans up the code quite a bit cleaner, so I'll work on this.